models/Recording.js

```diff
@@ -48,12 +49,38 @@ module.exports = function(sequelize, DataTypes) {
     instanceMethods: {
       canGetRaw: canGetRaw,
       getFileName: getFileName,
+      getUserPermissions: getUserPermissions,
     },
   };
 
   return sequelize.define(name, attributes, options);
 }
 
+/**
+ * Returns a JSON describing what the user can do to the recording.
+ * Premission types: DELETE, TAG, VIEW,
+ * //TODO This will be edited in the future when recordings can be public.
+ */
+function getUserPermissions(user) {
+  // For now if the user is in the group that owns the recording they have all
+  // permission. This will be changed in the future.
+  var permissions = {
+    canDelete: false,
+    canTag: false,
+    canView: false,
+  }
+  var recording = this;
+  return new Promise(async (resolve, reject) => {
+    var groupIds = await user.getGroupsIds();
+    if (groupIds.indexOf(recording.GroupId) !== -1) {
```

Comment:
What does it mean when the group id isn't in `groupsIds` and is it wise to give full permissions then?

Reply:
It will only give full permissions if the recording belongs to a group that the user is in. This can be changed in the future to limited permissions when we add custom group permissions.

Reply:
Sorry, I read it as "give permissions if the user isn't in the group" but that's obviously not it.